src/argparse.c

```diff
@@ -207,12 +362,37 @@ void neu_cli_args_init(neu_cli_args_t *args, int argc, char *argv[])
 
     memset(args, 0, sizeof(*args));
 
-    // load env
-    load_env(args, &log_level, &config_dir, &plugin_dir);
-
     int c            = 0;
     int option_index = 0;
 
+    // load config file
+
+    resolve_config_file_path(argc, argv, long_options, opts, &config_file);
+
+    if (config_file) {
```

Comment:
the `neu_cli_args_init` function is getting complex, maybe we could combine these `if` statements and the ` resolve_config_file_path` function into one ?